File: NeetCode - arrays and hashing/Challenge1.cpp
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        int size = nums.size();
        vector<int> ans;
        for(int x = 0; x < 2; ++x){
            for(int i = 0; i < size; ++i){
                ans.push_back(nums[i]);
            }
        }

        return ans;
    }
};


int main(){
    Solution sol;
    vector<int> b = {1, 2, 3};
    vector<int> a = sol.getConcatenation(b);

    for(int i : a){
        cout<<i<<" ";
    }


    return 0;
}

File: NeetCode - Binary Search/Challenge1.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        i = 0
        j = len(nums) - 1

        while i <= j:
            mid = int(i + (j - i) / 2)

            if nums[mid] < target:
                i = mid + 1
            
            elif nums[mid] > target:
                j = mid - 1
            
            else:
                return mid
        
        return -1

sol = Solution()
nums = [-1,0,2,4,6,8]
target = 4
print(sol.search(nums, target))